pkg/scheduler/metrics/metrics_record.go

```diff
@@ -0,0 +1,46 @@
+package metrics
+
+import "time"
+
+const (
+	scheduledResult = "scheduled"
+	errorResult     = "error"
+)
+
+const (
+	// BindingAdd is the event when a new binding is added to API server.
+	BindingAdd = "BindingAdd"
+	// BindingUpdate is the event when a new binding is updated to API server.
+	BindingUpdate = "BindingUpdate"
+	// ScheduleAttemptFailure is the event when a schedule attempt fails.
+	ScheduleAttemptFailure = "ScheduleAttemptFailure"
+	// PolicyChanged means binding needs to be rescheduled for the policy changed
+	PolicyChanged = "PolicyChanged"
+	// ClusterNotReady means binding needs to be rescheduled for cluster is not ready
+	ClusterNotReady = "ClusterNotReady"
+)
+
+// BindingSchedule can records a scheduling attempt and the duration
+// since `start`.
+func BindingSchedule(scheduleType string, duration float64, err error) {
+	if err != nil {
+		observeScheduleAttemptAndLatency(errorResult, scheduleType, duration)
+	} else {
+		observeScheduleAttemptAndLatency(scheduledResult, scheduleType, duration)
+	}
+}
+
+func observeScheduleAttemptAndLatency(result, scheduleType string, duration float64) {
+	e2eSchedulingLatency.WithLabelValues(result, scheduleType).Observe(duration)
+	scheduleAttempts.WithLabelValues(result, scheduleType).Inc()
+}
+
+// ScheduleStep can records each scheduling duration
+func ScheduleStep(action string, startTime time.Time) {
```

Comment:
Does the `action` means `step`? Should we define an enum for it?

Reply:
ok